src/csharp/Grpc.Core/Internal/PlatformApis.cs

```diff
@@ -33,12 +33,17 @@ namespace Grpc.Core.Internal
     internal static class PlatformApis
     {
         const string UnityEngineApplicationClassName = "UnityEngine.Application, UnityEngine";
+        const string XamarinAndroidApplicationClassName = "Android.App.Application, Mono.Android";
+        const string XamariniOSApplicationClassName = "UIKit.UIApplication, Xamarin.iOS";
```

Comment:
can you provide some reasoning why UIKit.UIApplication, Xamarin.iOS  is a good detection mechanism? I saw https://github.com/grpc/grpc/pull/15969/files is using a different class name (but using `Mono.CSharp.Enum, Mono.CSharp` actually might be a typo.

Reply:
Yes, `Mono.CSharp.Enum, Mono.CSharp` is wrong.

I'd suggest to go search for "Java.Lang.Object, Mono.Android" and "Foundation.NSObject, Xamarin.iOS" for Android and iOS, respectively.

Reply:
done here: https://github.com/grpc/grpc/pull/16099/files

Reply:
@ahouben That is a very good call! Thanks for pointing it out.